vqpy/utils/filters.py

```diff
@@ -0,0 +1,34 @@
+class continuing:
+    """A filter constraint that checks if the condition on property
+    is True for a certain duration
+
+    Returns True if the condition on `property` is True for the given
+    duration, with duration being counted accumulatively.
+    Attribute with name `property`_duration is used to store the
+    cumulative duration of condition being met.
+    """
+
+    def __init__(self, condition, duration, name):
+        self.condition = condition
+        self.duration = duration
+        # use name given as property name of duration
+        self.name = f"{name}_duration"
+
+    def __call__(self, obj, item):
+        if self.condition(obj.getv(item)):
+            # increment duration if `condition` returns True
+            found = False
+            # check frames from the last to the first
+            for i in range(-1, -obj._track_length, -1):
```

Comment:
The rewritten logic should be easier to read.
Time periods that exceeds the duration and potentially ongoing time periods are saved in VObj's `__static_` field, since they are not related to time (compared with `_datas`, which stores per-frame properties).

Reply:
Looks great!